client-state/models/Form.ts

```diff
@@ -53,10 +53,10 @@ export const Form = types
       const iComplete =
         incomeFields.length > 0 && incomeFields.every((field) => field.filled)
       const pComplete =
-        personalFields.length > 0 &&
+        personalFields.length > 0 && iComplete &&
         personalFields.every((field) => field.filled)
       const lComplete =
-        legalFields.length > 0 && legalFields.every((field) => field.filled)
+        legalFields.length > 0 && pComplete && legalFields.every((field) => field.filled)
```

Comment:
personal cannot be complete unless income is, and legal cannot be complete if the other two aren't